isaaclab_arena/tasks/sequential_task_base.py

```diff
@@ -0,0 +1,202 @@
+                continue
+            value = getattr(cfg_instance, field.name)
+            new_fields.append((field.name, field.type, value))
+            field_values[field.name] = value
+
+        if not new_fields:
+            return None
+
+        # Create a new configclass without the excluded fields
+        new_cfg_class = make_configclass(type(cfg_instance).__name__, new_fields)
+        return new_cfg_class(**field_values)
+
+    @staticmethod
+    def sequential_task_success_func(
+        env,
+        task_instance: "SequentialTaskBase",
+    ) -> torch.Tensor:
+        "Sequential task compositesuccess function."
+        # Check success of current subtask for each env
+        for env_idx in range(env.num_envs):
+            current_subtask_idx = env._current_subtask_idx[env_idx]
+            current_subtask_success_func = (
+                task_instance.subtasks[current_subtask_idx].get_termination_cfg().success.func
+            )
+            current_subtask_success_params = (
+                task_instance.subtasks[current_subtask_idx].get_termination_cfg().success.params
+            )
+            result = current_subtask_success_func(env, **current_subtask_success_params)[env_idx]
+
+            if result:
+                env._subtask_success_state[env_idx][current_subtask_idx] = True
+                if current_subtask_idx < len(task_instance.subtasks) - 1:
+                    env._current_subtask_idx[env_idx] += 1
+
+        # Compute composite task success state for each env
+        per_env_success = [all(env_successes) for env_successes in env._subtask_success_state]
+        success_tensor = torch.tensor(per_env_success, dtype=torch.bool, device=env.device)
+
+        env.extras["subtask_success_state"] = copy.copy(env._subtask_success_state)
+
+        return success_tensor
+
+    @staticmethod
+    def reset_subtask_success_state(
+        env,
+        env_ids,
+        task_instance: "SequentialTaskBase",
+    ) -> None:
+        "Reset subtask success vector and state machine for each environment."
+        # Initialize each env's subtask success state to False
+        if not hasattr(env, "_subtask_success_state"):
+            env._subtask_success_state = [[False for _ in task_instance.subtasks] for _ in range(env.num_envs)]
+        else:
+            for env_id in env_ids:
+                env._subtask_success_state[env_id] = [False for _ in task_instance.subtasks]
+
+        # Initialize each env's current subtask index (state machine) to 0
+        if not hasattr(env, "_current_subtask_idx"):
+            env._current_subtask_idx = [0 for _ in range(env.num_envs)]
+        else:
+            for env_id in env_ids:
+                env._current_subtask_idx[env_id] = 0
+
+    def get_scene_cfg(self) -> configclass:
+        "Make combined scene cfg from all subtasks."
+        scene_cfg = combine_configclass_instances("SceneCfg", *(subtask.get_scene_cfg() for subtask in self.subtasks))
```

Comment:
I wonder if we should add something in `isaaclab_arena.utils.configclass` to detect if there are repeated config class members. Here we could check and warn (before proceeding).